File: 2025-3/03.18/같은 숫자는 싫어.py
# I: 정수 배열 arr
# O: 연속된 중복을 제거한 배열
# C: 입력된 배열의 원소는 정수이다.
# E: 

# algo: 
# ds: 스택 
# solution:
# - 스택을 초기화한다.
# - 배열을 순회하면서 현재 값이 스택의 마지막 값과 다르면 스택에추가한다.
# - 최종적으로 중복이 제거된 배열을 반환한다.

def solution(arr):
    stack = [];
    
    for num in arr:
        if(len(stack) == 0):
            stack.append(num)
        if(stack[-1] != num):
            stack.append(num)
            
    return stack

File: 2025-03/03.20/같은숫자는싫어.js
/*
I:arr [1,1,3,3,0,1,1]
O:answer [1,3,0,1]
C:
배열 arr의 크기 : 1,000,000 이하의 자연수
배열 arr의 원소의 크기 : 0보다 크거나 같고 9보다 작거나 같은 정수

E: if(arr.length == 1)return arr;

algo: stack
ds: arr

solution
- 스택을 만들고
- for arr길이만큼
    - 현재 값과 스택 맨위게 같으면 걍 넘어가고
    - 아니면 스택에 추가한다
- 스택을 리턴한다.
*/

function solution(arr)
{
    var answer = [arr[0]];
    
    for(let i = 1; i < arr.length; i ++){
        if(answer[answer.length - 1] == arr[i])continue;
        answer.push(arr[i]);
    }
    
    return answer;
}